lib/ex_unit/lib/ex_unit/doc_test.ex

```diff
@@ -411,9 +412,13 @@ defmodule ExUnit.DocTest do
         quote do
           message = "Doctest did not compile, got: #{unquote(ex_message)}"
 
-          reraise ExUnit.AssertionError,
-                  [message: message, expr: unquote(String.trim(expr))],
-                  unquote(stack)
+          if unquote(String.valid?(expr)) do
```

Comment:
You can do this check at compilation time. So move the conditional out of the quote and then do something like:

    reraise ExUnit.AssertionError, unquote(opts), unquote(stack)

Please let me know if you have any questions. :)

Reply:
Got it 😉